src/renderers/dom/server/ReactServerUpdateQueue.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactServerUpdateQueue
+ * @flow
+ */
+
+'use strict';
+
+var ReactUpdateQueue = require('ReactUpdateQueue');
+var Transaction = require('Transaction');
+var ReactClass = require('ReactClass');
+var warning = require('warning');
+
+/**
+ * This is the server update queue. It uses ReactUpdateQueue while mounting
+ * then switches to ReactNoopUpdateQueue when mounted
+ * @class ReactServerUpdateQueue
+ * @param {boolean} transaction
```

Comment:
Yeah that's a good point, the constructor clearly shows that the transaction is of the Transaction type like 4 lines later. It's probably redundant. The only downside I could imagine is that people might be more used to parsing types from JSDoc annotations, but that's not a big concern since this isn't a public API